astropy/table/index.py

```diff
@@ -537,40 +524,36 @@ def replace(self, row, col, val):
         if not self._frozen:
             self.index.replace(self.orig_coords(row), col, val)
 
-    def copy(self):
+    def get_index_or_copy(self):
```

Comment:
Not for now, but this behaviour seems truly bizarre. At least with the name change it is slightly less confusing, but really no-one would expect that this influences the instance itself!